GenericNode/src/main/java/genericnode/Handlers/UdpClientHandler.java

```diff
@@ -0,0 +1,75 @@
+// Java
+
+package genericnode.Handlers;
+
+import genericnode.Servers.UDPServer;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.util.ArrayList;
+
+public class UdpClientHandler extends ClientHandler {
+
+    private final DatagramSocket socket;
+    private final DatagramPacket packet;
+
+    public UdpClientHandler(DatagramSocket socket, DatagramPacket packet, UDPServer udpServer) {
+        super(udpServer);
+        this.socket = socket;
+        this.packet = packet;
+    }
+
+    @Override
+    public void run() {
+        try {
+            String data = new String(packet.getData(), 0, packet.getLength());
+
+            // 3. Parse operation and key-value
+            String[] parts = data.split(" ");
+            String operation = parts[0];
+            String key = parts.length > 1 ? parts[1] : null;
+            String value = parts.length > 2 ? parts[2] : null;
```

Comment:
we can extract a new function out of it, just for clean code purposes, that way you wont have to write a comment. The function name itself should be self explanatory